iceoryx_posh/cmake/IceoryxPoshDeployment.cmake

```diff
@@ -27,46 +39,102 @@ if(NOT IOX_COMMUNICATION_POLICY)
     set(IOX_COMMUNICATION_POLICY ManyToManyPolicy)
 endif()
 
-if(NOT IOX_MAX_PUBLISHERS)
-    set(IOX_MAX_PUBLISHERS 512)
-endif()
-message(STATUS "[i] IOX_MAX_PUBLISHERS:" ${IOX_MAX_PUBLISHERS})
-
-if(NOT IOX_MAX_SUBSCRIBERS)
-    set(IOX_MAX_SUBSCRIBERS 1024)
-endif()
-message(STATUS "[i] IOX_MAX_SUBSCRIBERS:" ${IOX_MAX_SUBSCRIBERS})
-
-if(NOT IOX_MAX_INTERFACE_NUMBER)
-    set(IOX_MAX_INTERFACE_NUMBER 4)
-endif()
-message(STATUS "[i] IOX_MAX_INTERFACE_NUMBER:" ${IOX_MAX_INTERFACE_NUMBER})
-
-if(NOT IOX_MAX_SUBSCRIBERS_PER_PUBLISHER)
-    set(IOX_MAX_SUBSCRIBERS_PER_PUBLISHER 256)
-endif()
-message(STATUS "[i] IOX_MAX_SUBSCRIBERS_PER_PUBLISHER:" ${IOX_MAX_SUBSCRIBERS_PER_PUBLISHER})
-
-if(NOT IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY)
-    set(IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY 8)
+# Refer to iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/ipc_channel.hpp
+# for info why this is needed.
+if(APPLE)
+    set(IOX_MAX_RUNTIME_NAME_LENGTH_DEFAULT 98)
+else()
+    set(IOX_MAX_RUNTIME_NAME_LENGTH_DEFAULT 100)
 endif()
-message(STATUS "[i] IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY:" ${IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY})
 
-if(NOT IOX_MAX_PUBLISHER_HISTORY)
-    set(IOX_MAX_PUBLISHER_HISTORY 16)
-endif()
-message(STATUS "[i] IOX_MAX_PUBLISHER_HISTORY:" ${IOX_MAX_PUBLISHER_HISTORY})
-
-if(NOT IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY)
-    set(IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY 256)
-endif()
-message(STATUS "[i] IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY:" ${IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY})
+configure_option(
```

Comment:
OK, we keep the defaults from before here, right?

Reply:
Yes, these values are the defaults that were hard-coded before.